qiime2/core/type/signature.py

```diff
@@ -619,7 +619,8 @@ def _infer_type(self, key, value):
             inner = UnionExp(
                 (self._infer_type(key, v) for v in value.values()))
             return Collection[inner.normalize()]
-        if isinstance(value, qiime2.sdk.Artifact):
+        if isinstance(value, qiime2.sdk.Artifact) or \
```

Comment:
I think you can do 
`isinstance(value, (qiime2.sdk.Artifact, qiime2.sdk.proxy.ProxyArtifact))`